src/Data/Functor/Classes/Compat.hs

```diff
@@ -11,13 +14,27 @@ import Data.Functor.Classes
 #if !(MIN_VERSION_base(4,8,0))
 import Data.Monoid (mappend)
 #endif
+
+boringEq :: Eq1 f => f a -> f b -> Bool
+boringEq = liftEq (\_ _ -> True)
+
+emptyEq :: Eq1 f => f a -> f b -> Bool
+emptyEq = liftEq (\_ _ -> False)
+
+boringCompare :: Ord1 f => f a -> f b -> Ordering
+boringCompare = liftCompare (\_ _ -> EQ)
```

Comment:
It would be worth leaving some comments here to explain exactly what "boring" and "empty" mean in this context.